### PR TITLE
maintenance

### PR DESCRIPTION
**Pull request by nbdy**
as per commits:
- replaced deprecated protobuf library
- regenerated pb.go files
- dependency bumps
- adjusted code to [nnet changes](https://github.com/nknorg/nnet/pull/45)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,103 +1,106 @@
 module github.com/nknorg/nkn/v2
 
-go 1.19
+go 1.20
+
+replace github.com/nknorg/nnet => ../nnet
```
Comment (yilunzhang) [inline go.mod:5]: don't forget to remove this replace and upgrade nnet to the latest version that includes https://github.com/nknorg/nnet/pull/46 to be compatible with go 1.20

Comment (nbdy) [reply go.mod]: done. thanks for the reminder.

**Review by yilunzhang — APPROVED** [review]
LGTM

**Comment by yilunzhang** [comment]
I will fix github action failure in a separate PR

**Pull request merged**